File: services/admin-service/src/models/station.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Station {
    pub id: String,
    pub partner_id: Option<String>,
    pub name: String,
    pub address: Option<String>,
    pub location: String,
    pub deleted_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize)]
pub struct StationResponse {
    pub id: String,
    pub partner_id: Option<String>,
    pub name: String,
    pub address: Option<String>,
    pub latitude: f64,
    pub longitude: f64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Deserialize)]
pub struct CreateStationRequest {
    pub partner_id: String,
    pub name: String,
    pub address: Option<String>,
    pub latitude: f64,
    pub longitude: f64,
}

#[derive(Debug, Deserialize)]
pub struct UpdateStationRequest {
    pub name: Option<String>,
    pub address: Option<String>,
    pub latitude: Option<f64>,
    pub longitude: Option<f64>,
}

File: apps/test-dashboard/src/types/station.ts
import type { BaseEntity, TelemetrySnapshot } from './common';

export interface Station extends BaseEntity {
  partner_id: string;
  name: string;
  external_id: string;
  location: string;
  latitude: number;
  longitude: number;
  address: string;
  timezone: string;
  grid_limit_kw: number;
  deployed_at: string;
  deleted_at: string | null;

  /* aggregated */
  charger_count: number;
  chargers_active: number;
  total_power_kw: number;
  current_load_kw: number;
  telemetry: TelemetrySnapshot;
}

export interface CreateStationRequest {
  partner_id: string;
  name: string;
  location: string;
  latitude: number;
  longitude: number;
  address: string;
  timezone: string;
  grid_limit_kw: number;
}

export interface UpdateStationRequest {
  name?: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  address?: string;
  timezone?: string;
  grid_limit_kw?: number;
}
